pkg/localstore/gc.go

```diff
@@ -109,6 +117,7 @@ func (db *DB) collectGarbage() (collectedCount uint64, done bool, err error) {
 	done = true
 	first := true
 	start := time.Now()
+	candidates := make([]shed.Item, 0)
```

Comment:
I agree this will save on allocations, but then the cleanup would be ugly since we'd want to get rid of entries from the previous iteration, so in order to actually save the allocation we'd have to iterate over every item in the slice, setting it to `nil`, otherwise we'd have dangling `shed.Item`s in memory.

Reply:
cc @janos

Reply:
:) just  remember how many you  fill in, no need to go  through it

Reply:
I could work nicely, just to make sure that every time before the slice is sliced to zero length, it is iterated up to its length and set every element to "zero" Item. Zero item is the item with all slices as nil (important for memory usage) and integers as 0 (not important for memory usage). That way, no dangling Items will exist. Basically a type that would be `[]shed.Item` with `reset` method and `contains` method which would replace `swrm.Address.MemberOf` as per other @zelig's comment which I agree on, also. Somehow, this container naturally imposes a new type, even for future improvements as the data structure may change, as suggested in the description.

Reply:
so.... not sure if i understand but is this a blocker?

Reply:
> :) just remember how many you fill in, no need to go through it

This is incorrect, since you will have danging items on the last gc run after which the gc target has been met, leaving dangling `shed.Item`s in slice. We therefore have to iterate over the whole slice every time we finish GCing for those items to not be referenced anymore, making them available for garbage collection.

Reply:
@acud For the first iteration it is also fine as it is, for me.